test/test_fiber_worker_pool.rb

```
@@ -0,0 +1,252 @@
+    pool.enqueue(task_b, wrapper_b)
+
+    # Wait for both completions
+    2.times { completion_queue.pop }
+    elapsed = Time.now - start_time
+    pool.shutdown
+
+    assert wrapper_a.completed?
+    assert wrapper_b.completed?
+    # Should complete in ~0.1s (parallel), not ~0.2s (sequential)
+    assert elapsed < 0.2, "Parallel execution should complete in < 0.2s, took #{elapsed}s"
+  end
```

RuboCop (1.84.0)

[convention] Assignment Branch Condition size for `test_two_independent_tasks_run_in_parallel` is too high. [<10, 26, 1> 27.87/23]

(Metrics/AbcSize)